controller.go

```diff
@@ -144,20 +145,22 @@ func (c *healthCheckController) runServiceChecksByServiceNames(services map[stri
 	}).Checks
 
 	wg := sync.WaitGroup{}
+	tempCtx, cancel := context.WithCancel(context.Background())
```

Comment:
Is there a benefit of creating a temporary context over using the provided request one?

Reply:
The original context gets canceled midway through the iteration and it causes errors:
`cannot build health check result for services: failed to get the categories from kubernetes: client rate limiter Wait returned an error: context canceled `
That's why I've used this workaround